ui/src/components/role-policy/AddRuleFormForRole.js

```diff
@@ -91,14 +93,20 @@ export default class AddRuleFormForRole extends React.Component {
     }
 
     inputChanged(key, evt) {
-        this.setState({ [key]: evt.target.value });
-        this.props.onChange(key, evt.target.value);
+        if (key == 'case') {
+            this.setState({ [key]: !this.state.case });
```

Comment:
evt.target.checked can be used here

Reply:
Why use it? I'll still need to check the key

Reply:
setState is an asynchronous function, 
here you are setting state of 'case' by accessing the current state of 'case' which may not be updated from a previous call. so this can lead to wrong values being set and not showing up properly in UI.
This would be rare but still a possibility and setting the value using event object would be the best way.

Reply:
Got it. Thanks for the explanation!

Reply:
Fixed